WooCommerce/Classes/ViewRelated/Themes/ThemesCarouselViewModel.swift

```diff
@@ -0,0 +1,49 @@
+import Foundation
+import Yosemite
+
+/// View model for `ThemesCarouselView`
+///
+final class ThemesCarouselViewModel: ObservableObject {
+    enum State: Equatable {
+        case loading
+        case error
+        case content(themes: [WordPressTheme])
+    }
+
+    @Published private(set) var state: State = .loading
+
+    private let stores: StoresManager
+
+    init(stores: StoresManager = ServiceLocator.stores) {
+        self.stores = stores
+    }
+
+    @MainActor
+    func fetchThemes(currentThemeID: String? = nil) async {
```

Comment:
❓  Is it possible to move the logic of `currentThemeID` out of this method? 

This way we don't have to wait for the load current theme network request. i.e. We can load the current theme and load suggested themes in parallel. 

What do you think?

Reply:
I did this at first, but the theme list result arrives faster, so it'll be filtered after the current theme is loaded. This causes the filtered theme to appear briefly then disappear - which doesn't look nice. We can cache the theme name to UserDefaults so that the weird animation happens only on the first request, but it complicates the logic a bit more. WDYT @selanthiraiyan?

Reply:
Thanks for the explanation, Huong! 

I guess we have to separate the fetch and render process into separate methods to achieve a smooth experience. The extra complexity isn't worth it for the themes screen, considering we don't expect the user to access this theme setting often.